source/main.cpp

```diff
@@ -116,10 +116,11 @@ int main(int argc, char *argv[])
 	}
 	Files::Init(argv);
 
+	future<void> dataLoading;
 	try {
 		// Begin loading the game data.
 		bool isConsoleOnly = loadOnly || printShips || printTests || printWeapons;
-		GameData::BeginLoad(isConsoleOnly, debugMode);
+		dataLoading = GameData::BeginLoad(isConsoleOnly, debugMode);
 
 		// If we are not using the UI, or performing some automated task, we should load
 		// all data now. (Sprites and sounds can safely be deferred.)
```

Comment:
Should we be attempting to consume the future in these no-ui blocks?

Reply:
We could I guess. Should I? We'd need to call `wait_for(0s)` which looks a bit weird but not a big problem I guess.